build.go

```diff
@@ -63,9 +58,29 @@ func Build(
 
 		logger.SelectedDependency(entry, dependency, clock.Now())
 
+		if cachedSHA, ok := composerLayer.Metadata["dependency-sha"].(string); ok && cachedSHA == dependency.SHA256 {
+			logger.Process("Reusing cached layer %s", composerLayer.Path)
+			logger.Break()
+
+			composerLayer.Launch, composerLayer.Build = launch, build
```

Comment:
```suggestion
			composerLayer.Launch, composerLayer.Build, composerLayer.Cache = launch, build, build
```

Per [the spec](https://github.com/buildpacks/spec/blob/main/buildpack.md#layer-types), the `cache` flag makes the layer available on subsequent builds, which is what we'll need here I think. If you look at the table from the spec section I linked, when `build=true` and `cache=true`, the layer is restored from the cache, which is why we set ` composerLayer.Cache = build`. Confusingly, we don't need to check the `launch` layer for `cache` because when `launch=true`, the metadata will always be restored, telling the lifecycle to reuse the layer. It's confusing but the TL;DR is that we should set the `cache` flag here.

Reply:
Got it. I suspect the integration test just happens to pass due to lifecycle implementation, but really we should use `cache` to indicate that we want this layer cached.

https://paketobuildpacks.slack.com/archives/C011S6EL49L/p1649717116660929